From 5c5f1c05522ab9b6e7906c1f03fcb0905c7edbaa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marcin=20=C5=9Alusarz?= <marcin.slusarz@gmail.com>
Date: Sat, 9 Apr 2016 14:27:21 +0200
Subject: [PATCH] all: always append EXTRA_CFLAGS after our CFLAGS

...and add support for EXTRA_CFLAGS_DEBUG and EXTRA_CFLAGS_RELEASE.
---
 src/Makefile.inc           | 23 ++++++++++++-----------
 src/benchmarks/Makefile    |  3 ++-
 src/test/unittest/Makefile |  3 ++-
 src/tools/Makefile.inc     | 15 ++++++++-------
 4 files changed, 24 insertions(+), 20 deletions(-)

diff --git a/src/Makefile.inc b/src/Makefile.inc
index 798f514..efe7db7 100644
--- a/src/Makefile.inc
+++ b/src/Makefile.inc
@@ -55,7 +55,6 @@ CFLAGS += -Wconversion
 endif
 CFLAGS += -pthread
 CFLAGS += -fno-common
-CFLAGS += $(EXTRA_CFLAGS)
 CFLAGS += -DSRCVERSION=\"$(SRCVERSION)\"
 ifeq ($(call check_flag, -Wunreachable-code-return), y)
 CFLAGS += -Wunreachable-code-return
@@ -64,6 +63,18 @@ ifeq ($(call check_flag, -Wmissing-variable-declarations), y)
 CFLAGS += -Wmissing-variable-declarations
 endif
 
+ifeq ($(DEBUG),1)
+CFLAGS += -O0 -ggdb -DDEBUG $(EXTRA_CFLAGS_DEBUG)
+LIB_SUBDIR = /nvml_debug
+OBJDIR = debug
+else
+CFLAGS += -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 $(EXTRA_CFLAGS_RELEASE)
+LIB_SUBDIR =
+OBJDIR = nondebug
+endif
+
+CFLAGS += $(EXTRA_CFLAGS)
+
 LDFLAGS += -Wl,-z,relro -Wl,--fatal-warnings -Wl,--warn-common $(EXTRA_LDFLAGS)
 
 define arch32_error_msg
@@ -86,16 +97,6 @@ ifneq ($(LP64), 2)
 $(error $(arch32_error_msg))
 endif
 
-ifeq ($(DEBUG),1)
-CFLAGS += -O0 -ggdb -DDEBUG
-LIB_SUBDIR = /nvml_debug
-OBJDIR = debug
-else
-CFLAGS += -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2
-LIB_SUBDIR =
-OBJDIR = nondebug
-endif
-
 LIBS_DESTDIR = $(DESTDIR)$(libdir)$(LIB_SUBDIR)
 
 DIRNAME = $(shell basename $(CURDIR))
diff --git a/src/benchmarks/Makefile b/src/benchmarks/Makefile
index a10f23a..6e5d1c9 100644
--- a/src/benchmarks/Makefile
+++ b/src/benchmarks/Makefile
@@ -108,7 +108,6 @@ CFLAGS += -Wmissing-prototypes
 CFLAGS += -Wpointer-arith
 CFLAGS += -Wunused-macros
 CFLAGS += -pthread
-CFLAGS += $(EXTRA_CFLAGS)
 CFLAGS += -I../include
 CFLAGS += -I../libpmemobj
 CFLAGS += -I../common
@@ -140,6 +139,8 @@ ifeq ($(GLIB_SILENCE), y)
 CFLAGS += -Wno-unknown-attributes
 endif
 
+CFLAGS += $(EXTRA_CFLAGS)
+
 objdir=.
 
 %.o: %.c
diff --git a/src/test/unittest/Makefile b/src/test/unittest/Makefile
index 6b5f7bb..dc7ebd2 100644
--- a/src/test/unittest/Makefile
+++ b/src/test/unittest/Makefile
@@ -55,7 +55,6 @@ CFLAGS += -Wconversion
 endif
 CFLAGS += -pthread
 CFLAGS += -fno-common
-CFLAGS += $(EXTRA_CFLAGS)
 ifeq ($(call check_flag, -Wunreachable-code-return), y)
 CFLAGS += -Wunreachable-code-return
 endif
@@ -72,6 +71,8 @@ endif
 CFLAGS += -DUSE_LIBUNWIND
 endif
 
+CFLAGS += $(EXTRA_CFLAGS)
+
 all test: $(TARGET)
 
 $(TARGET): $(OBJS)
diff --git a/src/tools/Makefile.inc b/src/tools/Makefile.inc
index f3559d1..e036ef5 100644
--- a/src/tools/Makefile.inc
+++ b/src/tools/Makefile.inc
@@ -51,13 +51,6 @@ ifeq ($(call check_Wconversion), y)
 CFLAGS += -Wconversion
 endif
 CFLAGS += -fno-common
-CFLAGS += $(EXTRA_CFLAGS)
-
-ifeq ($(DEBUG),)
-CFLAGS += -O2 -D_FORTIFY_SOURCE=2
-else
-CFLAGS += -ggdb
-endif
 
 CFLAGS += -DSRCVERSION='"$(SRCVERSION)"'
 ifeq ($(call check_flag, -Wunreachable-code-return), y)
@@ -67,6 +60,14 @@ ifeq ($(call check_flag, -Wmissing-variable-declarations), y)
 CFLAGS += -Wmissing-variable-declarations
 endif
 
+ifeq ($(DEBUG),1)
+CFLAGS += -ggdb $(EXTRA_CFLAGS_DEBUG)
+else
+CFLAGS += -O2 -D_FORTIFY_SOURCE=2 $(EXTRA_CFLAGS_RELEASE)
+endif
+
+CFLAGS += $(EXTRA_CFLAGS)
+
 LDFLAGS += -Wl,-z,relro -Wl,--warn-common -Wl,--fatal-warnings $(EXTRA_LDFLAGS) -L$(TOP)/src/nondebug
 TARGET_DIR=$(DESTDIR)$(bindir)
 BASH_COMP_FILES ?=
-- 
2.5.0

